src/libraries/System.Formats.Tar/tests/TarReader/TarReader.GetNextEntry.Tests.cs

```diff
@@ -20,19 +20,6 @@ public void MalformedArchive_TooSmall()
             Assert.Throws<EndOfStreamException>(() => reader.GetNextEntry());
         }
 
-        [Fact]
-        public void MalformedArchive_HeaderSize()
```

Comment:
Why were these tests removed?

Reply:
because we are not throwing for invalid bytes anymore, following libarchive behavior.

Reply:
Right, but I am concerned that we now allow a whole header to contain invalid data. I do not think we want this level of flexiblity.

The test that is being removed would've thrown at some point due to having 512 consecutive 0x1 bytes. Let's analyze each one of the fields in the header:

From V7:
- name: This can be assumed to be valid since we store it as string and don't validate the value until we extract.
- mode, uid, guid, size, mtime, checksum: they all expect octal numbers. With your change, they would not throw if there are non-octal chars.
- linkname: Same as name.

From Ustar and Gnu:
- typeflag: We should be ok with any character here, even if it's not among the ones defined in our enum (this should throw right now, but it's a bug unrelated to this change, we should allow any ASCII character).
- magic: **this should throw**. We only have three possible values: either it's all nulls, or it's the Ustar magic, or it's the Gnu magic.
- version: Similar to magic.
- uname/gname: would not throw, we store this as a string.
- devmajor/devminor: Octal, this would pass with your change.
- prefix: Same as name and linkname.

Then since the size had some data, but we didn't find any characters, we would set size as 0 with your change. So no data.